src/common/types/value/value.cpp

```diff
@@ -513,16 +512,10 @@ Value::Value(const LogicalType& dataType_) : isNull_{true} {
     dataType = dataType_.copy();
 }
 
-void Value::copyFromFixedList(const uint8_t* fixedList) {
-    auto numBytesPerElement =
-        storage::StorageUtils::getDataTypeSize(*FixedListType::getChildType(dataType.get()));
-    for (auto i = 0u; i < childrenSize; ++i) {
-        auto childValue = children[i].get();
-        childValue->copyValueFrom(fixedList + i * numBytesPerElement);
-    }
-}
-
 void Value::copyFromVarList(ku_list_t& list, const LogicalType& childType) {
+    if (dataType->getLogicalTypeID() == LogicalTypeID::ARRAY) {
```

Comment:
Let's add a physical type `ARRAY` to avoid these checks.

Reply:
I think we should do it in another PR. I still have array using the same physical layout as var-list in this PR. And I removed the size check since we didn't have these checks for the old fixed-list vector.